File: Basic Math/Count Digits/CountDigits.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

class Solution {
  int evenlyDivides(int N) {
    int copy = N, counter = 0, rem = 0;
    while (N > 0) {
      rem = N % 10;
      if (rem != 0 && copy % rem == 0) counter++;
      N /= 10;
    }
    return counter;
  }
}

class CountDigits {
  public static void main(String args[]) throws IOException {
    BufferedReader read = new BufferedReader(new InputStreamReader(System.in));
    int t = Integer.parseInt(read.readLine());
    while (t-- > 0) {
      int N = Integer.parseInt(read.readLine());
      Solution ob = new Solution();
      System.out.println(ob.evenlyDivides(N));
      System.out.println("~");
    }
  }
}

File: Basic Math/Count Digits/CountDigits.cpp
#include <bits/stdc++.h>
using namespace std;
class Solution {
 public:
  int evenlyDivides(int N) {
    int copy = N, counter = 0, rem = 0;
    while (N > 0) {
      rem = N % 10;
      if (rem != 0 && copy % rem == 0) counter++;
      N /= 10;
    }
    return counter;
  }
};

int main() {
  int t;
  cin >> t;
  while (t--) {
    int N;
    cin >> N;
    Solution ob;
    cout << ob.evenlyDivides(N) << endl;

    cout << "~" << "\n";
  }
  return 0;
}